src/tests/features/rich_tcp_mss_clamp.at

```diff
@@ -0,0 +1,81 @@
+dnl ###############
+dnl ## XML tests ##
+dnl ###############
+
+FWD_START_TEST([tcp-mss-clamp])
+AT_KEYWORDS(tcp-mss-clamp)
+
+AT_CHECK([mkdir -p ./zones])
+
+AT_DATA([./zones/tcp.xml], [dnl
+<?xml version="1.0" encoding="utf-8"?>
+<zone>
+  <service name="ssh"/>
+  <service name="dhcpv6-client"/>
+  <rule>
+    <tcp-mss-clamp value="536"/>
+    <accept/>
+  </rule>
+</zone>
+])
+FWD_CHECK([--check-config],0, ignore)
+
+AT_DATA([./zones/tcp.xml], [dnl
+<?xml version="1.0" encoding="utf-8"?>
+<zone>
+  <service name="ssh"/>
+  <service name="dhcpv6-client"/>
+  <rule>
+    <tcp-mss-clamp value="pmtu"/>
+    <accept/>
+  </rule>
+</zone>
+])
+FWD_CHECK([--check-config],0, ignore)
+
+AT_DATA([./zones/tcp.xml], [dnl
+<?xml version="1.0" encoding="utf-8"?>
+<zone>
+  <service name="ssh"/>
+  <service name="dhcpv6-client"/>
+  <rule>
+    <tcp-mss-clamp value="thisdoesnotexist"/>
+    <accept/>
+  </rule>
+</zone>
+])
+m4_ifdef([TESTING_FIREWALL_OFFLINE_CMD], [FWD_CHECK([--check-config], 0, ignore, [dnl
+WARNING: INVALID_RULE: thisdoesnotexist: rule tcp-mss-clamp value="thisdoesnotexist" accept
+WARNING: INVALID_RULE: thisdoesnotexist: rule tcp-mss-clamp value="thisdoesnotexist" accept
+])], [FWD_CHECK([--check-config], 0, ignore, [dnl
+])])
```

Comment:
Since the "online" tests cause the warning to be printed to the log file you can check the expected warning get logged with the macro `FWD_GREP_LOG`. This is only needed since the error code is wrong (0).

Reply:
Firewalld doesn't print the warning to the log file unless the debug flag is enabled. This means that without the --debug flag FWD_GREP_LOG fails. Do you have any suggestions on how to get around this? I tested out a number of things and the only way I know how to resolve this issue is reverting back to the code snippet above.

Reply:
Hrm. I thought WARNING logs were always printed. Even without `--debug`

Reply:
The problem is actually because you're not calling `--check-config` before `FWD_GREP_LOG`.